spatialos-sdk/src/worker/schema/component_update.rs

```diff
@@ -0,0 +1,185 @@
+use crate::worker::{
+    component::{Component, ComponentId, Update},
+    schema::{owned::*, ArrayField, FieldId, Object, SchemaField, SchemaObjectType},
+};
+use spatialos_sdk_sys::worker::*;
+use std::marker::PhantomData;
+
+#[derive(Debug)]
+pub struct ComponentUpdate(PhantomData<*mut Schema_ComponentUpdate>);
+
+impl ComponentUpdate {
+    pub fn new<U: Update>(update: &U) -> Owned<Self> {
+        // Create the underlying `Schema_ComponentUpdate` and wrap it in a smart pointer.
+        let mut result: Owned<Self> =
+            unsafe { Owned::new(Schema_CreateComponentUpdate(U::Component::ID)) };
+
+        // Populate the update object from the update data.
+        let component_update = &mut *result;
+        update.into_update(component_update);
+
+        result
+    }
+
+    pub(crate) unsafe fn from_raw<'a>(raw: *mut Schema_ComponentUpdate) -> &'a Self {
+        &*(raw as *mut _)
+    }
+
+    pub fn component_id(&self) -> ComponentId {
+        unsafe { Schema_GetComponentUpdateComponentId(self.as_ptr()) }
```

Comment:
Note that `Schema_GetComponentUpdateComponentId` (and friends) is gone in 14.0.0. You should instead get the component ID (and command index for commands) from the oplist when the `Worker_ComponentUpdate`/`Worker_CommandRequest` is given to you, and passed along to the required functions. Perhaps `from_raw` can take a component ID and command index (where applicable).